### PR TITLE
[Core] Add back ray queue to put request output tokens back to the api server

### PR DESCRIPTION
**Pull request by KuilongCui**

### CODE REVIEW

File: llumnix/backends/vllm/llm_engine.py
Comment (s5u13b) [inline llumnix/backends/vllm/llm_engine.py:45]: We can give data type hint, instance: str, output_queue_type: xxx...

File: llumnix/backends/vllm/llm_engine.py
Comment (s5u13b) [inline llumnix/backends/vllm/llm_engine.py:102]: enum type is better

File: llumnix/output_queue/output_queue_client_base.py
Comment (s5u13b) [inline llumnix/output_queue/output_queue_client_base.py:1]: queue_client_base, queue_server_base is enough. Also, for directory, output_queue -> queue, because although we use queue to put/get request output token, but the queue and its interface is general, not specific to request output.

File: .github/workflows/bench_test.yml
Comment (s5u13b) [inline .github/workflows/bench_test.yml:4]: I think we should test when push to main, because the codes might change after merging to main.

File: configs/base.yml
Comment (s5u13b) [inline configs/base.yml:12]: if the default migration backend is gloo, they both should be false.

File: llumnix/entrypoints/vllm/api_server.py
Comment (s5u13b) [inline llumnix/entrypoints/vllm/api_server.py:255]: We can test arguments in arguments checker.

**Review by s5u13b — APPROVED** [review]

File: configs/base.yml
Comment (KuilongCui) [reply configs/base.yml:12]: the default migration backend is rpc now.

File: llumnix/backends/vllm/llm_engine.py
Comment (KuilongCui) [reply llumnix/backends/vllm/llm_engine.py:45]: get

File: llumnix/backends/vllm/llm_engine.py
Comment (KuilongCui) [reply llumnix/backends/vllm/llm_engine.py:102]: leave it to next step
mark it as a TODO

File: .github/workflows/bench_test.yml
Comment (KuilongCui) [reply .github/workflows/bench_test.yml:4]: Directly pushing to the main branch is forbidden; only pull requests are allowed. Code can only be merged into the main branch once all continuous integration checks have passed.

File: llumnix/entrypoints/vllm/api_server.py
Comment (KuilongCui) [reply llumnix/entrypoints/vllm/api_server.py:255]: For arguments with specified choices, the argument checker will automatically validate the values.

File: llumnix/output_queue/output_queue_client_base.py
Comment (KuilongCui) [reply llumnix/output_queue/output_queue_client_base.py:1]: get

File: llumnix/entrypoints/vllm/api_server.py
Comment (KuilongCui) [reply llumnix/entrypoints/vllm/api_server.py:255]: see llumnix/arg_utils.py::check_args

**Review by ZeldaHuang — APPROVED** [review]

File: configs/base.yml
Comment (s5u13b) [reply configs/base.yml:12]: We can set default migration backend as 'gloo' and set these two default values as False.

File: examlpes/offline_inference.py
Comment (s5u13b) [inline examlpes/offline_inference.py:51]: Use enum as request_output_queue type?

**Pull request merged**
